src/pages/index.js

```diff
@@ -1,168 +1,241 @@
+const profileEditButton = document.querySelector(".profile__edit-button");
+const deleteCardPopup = new PopupWithConfirmation("#modal__delete");
+const previewImagePopup = new PopupWithImage(
+  selectors.previewModal,
+  handleCardClick
 );
-
-section.renderItems();
-
-function createCard(cardData) {
-  const card = new Card(cardData, selectors.cardTemplate, handleCardClick);
-
-  return card.getView();
-}
-
-function handleCardClick(cardData) {
-  previewImagePopup.open(cardData);
-}
-
-/* -------------------------------------------------------------------------- */
-
-/*                                  USER INFO                                 */
-
-/* -------------------------------------------------------------------------- */
-
+const profileEditPopup = new PopupWithForm(
+  selectors.profilePopupSelector,
+  handleProfileEditSubmit
+);
+const addCardPopup = new PopupWithForm(
+  selectors.cardPopupSelector,
+  handleCardAddClick
+);
+const cardSelector = selectors.cardTemplate;
 const userInfo = new UserInfo({
   userNameSelector: ".profile__title",
-
   userDescriptionSelector: ".profile__description",
+  userAvi: ".profile__image",
 });
 
 /* -------------------------------------------------------------------------- */
-
-/*                              POPUP WITH IMAGE                              */
-
+/*                                  FUNCTIONS & HANDLERS                      */
 /* -------------------------------------------------------------------------- */
-const previewImagePopup = new PopupWithImage("#modal-preview");
-previewImagePopup.setEventListeners();
+let newCardSection;
+let userId;
 
-/* -------------------------------------------------------------------------- */
-
-/*                               POPUP WITH FORM                              */
-
-/* -------------------------------------------------------------------------- */
-
-/* --------------------------- edit profile modal --------------------------- */
-
-const profileEditButton = document.querySelector(".profile__edit-button");
-
-const profileEditPopup = new PopupWithForm(
-  "#profile-edit-modal",
-
-  handleProfileEditSubmit
-);
-
-profileEditPopup.setEventListeners();
+function createCard(cardData) {
+  const card = new Card(
+    cardData.name,
+    cardData.link,
+    cardData.isLiked,
+    cardData.likes,
+    cardData._id,
+    userId,
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick,
+    cardSelector
+  );
 
-profileEditButton.addEventListener("click", () => {
-  handleProfileEditClick();
-});
+  return card.getView();
+}
+function handleProfileEditSubmit(inputValues) {
+  profileEditPopup.renderLoading(true);
+  api
+    .updateProfileInfo(inputValues)
+    .then(() => {
+      userInfo.setUserInfo(inputValues);
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      profileEditPopup.renderLoading(false, "Save");
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```